applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,555 @@
+		if (ret < 0) {
+			LOG_ERR("Load ringbuf err: %d", ret);
+			break;
+		}
+
+		if (i == 0) {
+			/* Data can now be read from the ringbuffer */
+			sd_card_playback_active = true;
+		}
+	}
+
+	sd_card_playback_active = false;
+
+	ret_sd_card_close = sd_card_close(&f_seg_read_entry);
+	/* Check if something inside the for loop failed */
+	if (ret) {
+		LOG_ERR("WAV playback err: %d", ret);
+		return ret;
+	}
+
+	if (ret_sd_card_close) {
+		LOG_ERR("SD card close err: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int sd_card_playback_play_lc3(void)
+{
+	int ret;
+	int ret_sd_card_close;
+	uint16_t pcm_mono_write_size;
+	uint8_t decoder_num_ch = audio_system_decoder_num_ch_get();
+	size_t lc3_file_header_size = sizeof(lc3_file_header);
+	size_t lc3_frame_header_size = sizeof(uint16_t);
+
+	ret = sd_card_open(playback_file_name, &f_seg_read_entry);
+	if (ret) {
+		LOG_ERR("Open SD card file err: %d", ret);
+		return ret;
+	}
+
+	/* Read the file header */
+	ret = sd_card_read((char *)&lc3_file_header, &lc3_file_header_size, &f_seg_read_entry);
+	if (ret < 0) {
+		LOG_ERR("Read SD card file err: %d", ret);
+		return ret;
+	}
+
+	pcm_frame_size = sizeof(uint16_t) * lc3_file_header.sample_rate *
+			 lc3_file_header.frame_duration / 1000;
+	lc3_playback_cfg.lc3_frames_num =
+		sizeof(uint16_t) *
+		((lc3_file_header.signal_len_msb << 16) + lc3_file_header.signal_len_lsb) /
+		pcm_frame_size;
+
+	uint8_t pcm_mono_frame[pcm_frame_size];
+
+	sd_card_playback_active = true;
+	for (int i = 0; i < lc3_playback_cfg.lc3_frames_num; i++) {
+		/* Read the frame header */
+		ret = sd_card_read((char *)&lc3_playback_cfg.lc3_frame_length_bytes,
+				   &lc3_frame_header_size, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			sd_card_playback_active = false;
+			break;
+		}
+
+		char lc3_frame[lc3_playback_cfg.lc3_frame_length_bytes];
+		size_t lc3_fr_len = lc3_playback_cfg.lc3_frame_length_bytes;
+
+		/* Read the audio data frame to be decoded */
+		ret = sd_card_read(lc3_frame, &lc3_fr_len, &f_seg_read_entry);
+		if (ret < 0) {
+			LOG_ERR("SD card read err: %d", ret);
+			sd_card_playback_active = false;
+			break;
+		}
+
+		if (lc3_fr_len != lc3_playback_cfg.lc3_frame_length_bytes) {
+			LOG_ERR("Read size (%d) not equal requested size (%d)", lc3_fr_len,
+				lc3_playback_cfg.lc3_frame_length_bytes);
+			ret = -EPERM;
+			break;
+		}
+
+		/* Decode audio data frame */
+		ret = sw_codec_lc3_dec_run(lc3_frame, lc3_playback_cfg.lc3_frame_length_bytes,
+					   pcm_frame_size, decoder_num_ch - 1, pcm_mono_frame,
+					   &pcm_mono_write_size, false);
+		if (ret) {
+			LOG_ERR("Decoding err: %d", ret);
+			sd_card_playback_active = false;
+			break;
+		}
+
+		ret = sd_card_playback_ringbuf_write((char *)pcm_mono_frame, pcm_mono_write_size);
+		if (ret < 0) {
+			LOG_ERR("Load ringbuf err: %d", ret);
+			sd_card_playback_active = false;
+			break;
+		}
+	}
+
```

Comment:
This got a bit messy with back and forth - but you can set `sd_card_playback_active` to `false` here in the same way you're doing it for WAV so they're as similar as possible